python/cudf/cudf/core/tools/datetimes.py

```diff
@@ -814,23 +820,11 @@ def date_range(
     if isinstance(freq, DateOffset):
         offset = freq
     elif isinstance(freq, str):
-        # Map pandas `offset alias` into cudf DateOffset `CODE`, only
-        # fixed-frequency, non-anchored offset aliases are supported.
-        mo = re.fullmatch(
-            rf'(-)*(\d*)({"|".join(_offset_alias_to_code.keys())})', freq
-        )
-        if mo is None:
+        offset = DateOffset._from_str(freq)
+        if "months" in offset.kwds or "years" in offset.kwds:
             raise ValueError(
```

Comment:
it seems like `DateOffset._from_freqstr` raises this error now (the pattern failed to match completely). Should this error be tweaked to say something specific about non fixed frequency inputs?